internal/web/validate.go

```diff
@@ -74,11 +74,183 @@ func handleValidationConfig(cfgpath string) (Validationcfg, error) {
+			}
+			validateNifti = valcfg.Bidscfg.ValidateNifti
+		} else {
+			log.ShowWrite("[Error] unmarshalling validation config file: %s", err.Error())
+		}
+	} else {
+		log.ShowWrite("[Info] no validation config file found or processed, running from repo root (%s)", err.Error())
+	}
+
+	// Ignoring NiftiHeaders for now, since it seems to be a common error
+	outBadge := filepath.Join(resdir, srvcfg.Label.ResultsBadge)
+	log.ShowWrite("[Info] Running bids validation: '%s %t --json %s'", srvcfg.Exec.BIDS, validateNifti, valroot)
+
+	// Make sure the validator arguments are in the right order
+	var args []string
+	if !validateNifti {
+		args = append(args, "--ignoreNiftiHeaders")
+	}
+	args = append(args, "--json")
+	args = append(args, valroot)
+
+	// cmd = exec.Command(srvcfg.Exec.BIDS, validateNifti, "--json", valroot)
+	var out, serr bytes.Buffer
+	cmd := exec.Command(srvcfg.Exec.BIDS, args...)
+	out.Reset()
+	serr.Reset()
+	cmd.Stdout = &out
+	cmd.Stderr = &serr
+	// cmd.Dir = tmpdir
+	if err := cmd.Run(); err != nil {
+		log.ShowWrite("[Error] running bids validation (%s): '%s', '%s', '%s'",
+			valroot, err.Error(), serr.String(), "")
+
+		err = ioutil.WriteFile(outBadge, []byte(resources.FailureBadge), os.ModePerm)
+		if err != nil {
+			log.ShowWrite("[Error] writing results badge for %q", valroot)
+		}
+		// return err
+	}
+
+	// We need this for both the writing of the result and the badge
+	output := out.Bytes()
+
+	// CHECK: can this lead to a race condition, if a job for the same user/repo combination is started twice in short succession?
+	outFile := filepath.Join(resdir, srvcfg.Label.ResultsFile)
+	err := ioutil.WriteFile(outFile, []byte(output), os.ModePerm)
+	if err != nil {
+		log.ShowWrite("[Error] writing results file for %q", valroot)
+	}
+
+	// Write proper badge according to result
+	content := resources.SuccessBadge
+	var parseBIDS BidsRoot
+	err = json.Unmarshal(output, &parseBIDS)
+	if err != nil {
+		log.ShowWrite("[Error] unmarshalling results json: %s", err.Error())
+		content = resources.FailureBadge
+	} else if len(parseBIDS.Issues.Errors) > 0 {
+		content = resources.FailureBadge
+	} else if len(parseBIDS.Issues.Warnings) > 0 {
+		content = resources.WarningBadge
+	}
+
+	err = ioutil.WriteFile(outBadge, []byte(content), os.ModePerm)
+	if err != nil {
+		log.ShowWrite("[Error] writing results badge for %q", valroot)
+		// return err
+	}
+
+	log.ShowWrite("[Info] finished validating repo at %q", valroot)
+	return nil
+}
+
+// validateNIX runs the NIX validator on the specified repository in 'path'
+// and saves the results to the appropriate document for later viewing.
+func validateNIX(valroot, resdir string) error {
+	srvcfg := config.Read()
+
+	// TODO: Allow validator config that specifies file paths to validate
+	// For now we validate everything
+	nixfiles := make([]string, 0)
+	// Find all NIX files (.nix) in the repository
+	nixfinder := func(path string, info os.FileInfo, err error) error {
```

Comment:
This function always returns `nil` and never an actual `error`.

Reply:
The signature of the walker function is determined by `filepath.Walk()` so this can't change.